pywebpack/__init__.py

```diff
@@ -7,7 +7,289 @@
+        project_template_dir='buildconfig',  # `webpack.config.js` location
+        config={'debug': True},
+        config_path='build/config.json',
+    )
+
+``debug: True`` is an example of configuration that can be injected from Python
+and expose it to Webpack via the generated ``config.json``.
+
+Assets for multiple modules
+---------------------------
+
+It is a common practice to split your assets in several files to decrease
+assets file sizes and speed up webpage loading. Pywebpack lets you define a
+project that can have bundles, which are definitions of assets and npm
+dependencies.
+
+The recommended folder structure for a project with modules is the following::
+
+    /buildconfig
+        /package.json
+        /webpack.config.js
+    /modules
+        /module1
+            /...
+            /static
+                /js
+                /css
+        /module2
+            /...
+            /static
+                /js
+                /css
+    main.py  # or any script name
+
+Pywebpack will copy all bundles static files in the same working directory.
+It is **important** to name assets with distinct filenames, or create a
+namespaced subfolder to contain your static files, so that name collisions
+will be avoided.
+
+In our main script, let's add two bundles (called ``mainsite`` and
+``backoffice``)::
+
+    from pywebpack.bundle import WebpackBundle
+    mainsite = WebpackBundle(
+        './modules/mainsite/static',
+        entry={
+            'mainsite-base': './js/mainsite-base.js',
+            'mainsite-products': './js/mainsite-products.js',
+        },
+        dependencies={
+            'jquery': '^3.2.1'
+        }
+    )
+
+    backoffice = WebpackBundle(
+        './modules/backoffice/static',
+        entry={
+            'backoffice-base': './js/backoffice-base.js',
+            'backoffice-admin': './js/backoffice-admin.js',
+        },
+        dependencies={
+            'jquery': '^3.1.0'
+        }
+    )
+
+A :class:`~pywebpack.bundle.WebpackBundle` requires the path to the static
+files of the module, an entry for each asset with its relative path and any
+extra npm package.
+
+Then, we create a project :class:`~pywebpack.project.WebpackBundleProject`
+for our bundles::
+
+    from pywebpack.project import WebpackBundleProject
+    project = WebpackBundleProject(
+        working_dir='build',  # where config and assets files will be copied
+        project_template_dir='buildconfig',  #`webpack.config.js` location
+        bundles=[mainsite, backoffice]
+    )
+
+Pywebpack will generate a ``config.json`` which will contain all the assets
+with their paths. Each bundle entry will be a `webpack entry`_.
+
+As last step, in our ``webpack.config.js``, we set the entry to the
+``config.json`` so webpack will build each asset::
+
+    var path = require('path');
+    var config = require('./config')
+
+    module.exports = {
+        context: path.resolve(__dirname),
+        entry: config.entry,
```

Comment:
``// Entries from mainsite and backoffice bundles.``

When reading the example, it's a little unclear to me how bundle entries make it into the webpack config. Perhaps adding above comment is enough. Alternatively, we could show the generated config.json.